### PR TITLE
✨ add tag component

### PR DESCRIPTION
**Pull request by okusindja**

### CODE REVIEW

**Review by git-marcopitra — COMMENTED** [review]
Need fix

File: src/lib/components/tag/index.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    <MotionTag
+      {...(props.variant === 'filled' || props.variant === 'outline'
+        ? {
+            py:
+              size === 'large' ? '2xs' : size === 'medium' ? '2xs' : '.125rem',
+            px:
+              size === 'large' ? '2xs' : size === 'medium' ? '2xs' : '.125rem',
+          }
+        : props.variant === 'outline'
+        ? {
```
Comment (KipandaJr) [inline src/lib/components/tag/index.tsx:37]: Why you use a !important?

File: src/lib/components/tag/index.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        scale: props.disabled ? 1 : 1.05,
+        transition: { duration: 0.005, ease: easeInOut },
+      }}
+      {...props}
+    >
+      {PrefixIcon && (
+        <Box
+          p=".1875rem"
+          width={
+            size === 'large' ? '2rem' : size === 'medium' ? '1.8rem' : '1.3rem'
```
Comment (KipandaJr) [inline src/lib/components/tag/index.tsx:61]: we didnt has this sizes on theme?

File: src/lib/components/tag/index.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  variant({
+    scale: 'tags',
+    property: 'variant',
+  })
+);
+
+const MotionTag = motion(TagElement) as CustomDomComponent<TagProps>;
+
+export const Tag: FC<PropsWithChildren<TagProps>> = ({
+  onClose,
```
Comment (KipandaJr) [inline src/lib/components/tag/index.tsx:20]: you can dispense with the use of hasCloseIcon by just having `onClose?: () => void();` and to check whether or not you will have the close icon, you can validate whether the `onClose && <DoingThings />` function exists, if it does then it has an icon, and you avoid using one too many props.

File: src/lib/theme/dark/variants/tags.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    },
+    ':focus': {
+      background: colors.primary,
+    },
+    ':active': {
+      color: 'white',
+      background: colors.primary,
+    },
+    ':disabled': {
+      background: colors['outlineVariant'],
```
Comment (KipandaJr) [inline src/lib/theme/dark/variants/tags.ts:50]: use only one pattern to call the colors, `colors['outlineVariant']` ou `colors.outlineVariant`

File: src/lib/components/tag/index.tsx (hunk 2 above)
Comment (okusindja) [reply src/lib/components/tag/index.tsx:61]: these spaces? no we don't have these spaces values

File: src/lib/components/tag/index.tsx (hunk 1 above)
Comment (okusindja) [reply src/lib/components/tag/index.tsx:37]: I'll check this

File: src/lib/components/tag/index.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  hasCloseIcon,
+  size = 'large',
+  ...props
+}) => {
+  return (
+    <MotionTag
+      {...(props.variant === 'filled' || props.variant === 'outline'
+        ? {
+            py:
+              size === 'large' ? '2xs' : size === 'medium' ? '2xs' : '.125rem',
```
Comment (MarioBatalha) [inline src/lib/components/tag/index.tsx:30]: Is your intention to keep size 2xs as default? To skip this redundancy, you can do it this way:  `size ? '2xs' : size === 'medium' ? '2xs' : .125rem`. But note that the size will be 2xs as default

**Review by KipandaJr — APPROVED** [review]

File: src/lib/theme/light/variants/tags.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    background: colors['outlineVariant'],
+  },
+});
+
+export const variants = {
+  filled: mergeDeepRight(defaultTag, {
+    color: colors.onSurface,
+    background: '#fff',
+    ':hover': {
+      background: '#00000014',
```
Comment (MarioBatalha) [inline src/lib/theme/light/variants/tags.ts:20]: As suggestion, you can add some transition in hover and focus

File: src/lib/components/tag/index.tsx (hunk 5 above)
Comment (okusindja) [reply src/lib/components/tag/index.tsx:30]: @MarioBatalha The default size is 'Large' which is initialized in the props above

**Review by git-marcopitra — APPROVED** [review]

**Review by MarioBatalha — APPROVED** [review]

**Pull request merged**